packages/valory/connections/ledger/tests/test_ledger_api.py

```diff
@@ -78,6 +97,75 @@
     ],
 )
 
+ETHEREUM_DEFAULT_LEDGER_CONFIG = {
+    "address": f"{DEFAULT_GANACHE_ADDR}:{DEFAULT_GANACHE_PORT}",
+    "chain_id": DEFAULT_GANACHE_CHAIN_ID,
+    # "denom": ETHEREUM_DEFAULT_CURRENCY_DENOM, # noqa: E800
+    # "gas_price_api_key": GAS_PRICE_API_KEY, # noqa: E800
+    "default_gas_price_strategy": "eip1559",
+    "gas_price_strategies": {
+        "gas_station": DEFAULT_GAS_STATION_STRATEGY,
+        "eip1559": DEFAULT_EIP1559_STRATEGY,
+    },
+}
+
+
+@pytest.fixture(scope="session")
+def ethereum_testnet_config() -> Dict:
+    """Get Ethereum ledger api configurations using Ganache."""
+    new_uri = f"{DEFAULT_GANACHE_ADDR}:{DEFAULT_GANACHE_PORT}"
+    new_config = ETHEREUM_DEFAULT_LEDGER_CONFIG
+    new_config["address"] = new_uri
+    return new_config
+
+
+@pytest.fixture(scope="function")
+def update_default_ethereum_ledger_api(ethereum_testnet_config: Dict) -> Generator:
+    """Change temporarily default Ethereum ledger api configurations to interact with local Ganache."""
+    old_config = DEFAULT_LEDGER_CONFIGS.pop(EthereumCrypto.identifier)
+    DEFAULT_LEDGER_CONFIGS[EthereumCrypto.identifier] = ethereum_testnet_config
+    yield
+    DEFAULT_LEDGER_CONFIGS.pop(EthereumCrypto.identifier)
+    DEFAULT_LEDGER_CONFIGS[EthereumCrypto.identifier] = old_config
+
+
+def make_ledger_api_connection(
+    ethereum_testnet_config: Dict = ETHEREUM_DEFAULT_LEDGER_CONFIG,
+) -> Connection:
+    """Make a connection."""
+    crypto = make_crypto(DEFAULT_LEDGER)
+    identity = Identity("name", crypto.address, crypto.public_key)
+    crypto_store = CryptoStore()
+    directory = PACKAGE_DIR
+    connection = Connection.from_dir(
+        str(directory),
+        data_dir=MagicMock(),
+        identity=identity,
+        crypto_store=crypto_store,
+    )
+    connection = cast(Connection, connection)
+    connection._logger = logging.getLogger("packages.valory.connections.ledger")
+
+    # use testnet config
+    connection.configuration.config.get("ledger_apis", {})[
+        "ethereum"
+    ] = ethereum_testnet_config
+
+    connection.request_retry_attempts = 1  # type: ignore
+    connection.request_retry_attempts = 2  # type: ignore
+    return connection
+
+
+@pytest.fixture()
+async def ledger_apis_connection(
+    request: Any, ethereum_testnet_config: Dict
+) -> AsyncGenerator:
+    """Make a connection."""
+    connection = make_ledger_api_connection(ethereum_testnet_config)
+    await connection.connect()
+    yield connection
+    await connection.disconnect()
```

Comment:
So this might introduce some code repetition since we need to define these fixtures in multiple places. Once we're done and satisfied with the state of the package tests we can start extracting these functionalities to their respective plugins or test tools modules